File: Low Level Design (LLD)/DesignPatterns/1. Creational/Builder/Java/LenovoDesktopBuilder.java
public class LenovoDesktopBuilder extends DesktopBuilder {

    @Override
    public void buildMonitor() {
        desktop.setMonitor("Lenovo 27-inch IPS");
    }

    @Override
    public void buildMouse() {
        desktop.setMouse("Lenovo Ergonomic Mouse");
    }

    @Override
    public void buildKeyboard() {
        desktop.setKeyboard("Lenovo ThinkPad Keyboard");
    }

    @Override
    public void buildSpeaker() {
        desktop.setSpeaker("Dolby Audio Speakers");
    }

    @Override
    public void buildRam() {
        desktop.setRam("32GB DDR5");
    }

    @Override
    public void buildProcessor() {
        desktop.setProcessor("AMD Ryzen 9");
    }

    @Override
    public void buildMotherboard() {
        desktop.setMotherboard("Lenovo Legion Pro");
    }
}

File: Low Level Design (LLD)/DesignPatterns/1. Creational/Builder/Cpp/src/LenovoDesktopBuilder.cpp
#include "../include/LenovoDesktopBuilder.hpp"

// Build the monitor
void LenovoDesktopBuilder::buildMonitor() {
    desktop->setMonitor("Lenovo 1440p 27-inch 144hz IPS");
};

// Build the mouse
void LenovoDesktopBuilder::buildMouse() {
    desktop->setMouse("Lenovo Ergonomic Mouse");
};

// Build the keyboard
void LenovoDesktopBuilder::buildKeyboard() {
    desktop->setKeyboard("Lenovo Mechanical Keyboard");
};

// Build the speaker
void LenovoDesktopBuilder::buildSpeaker() {
    desktop->setSpeaker("Dolby Audio Speakers");
};

// Build the RAM
void LenovoDesktopBuilder::buildRam() {
    desktop->setRam("24GB DDR5");
};

// Build the processor
void LenovoDesktopBuilder::buildProcessor() {
    desktop->setProcessor("Intel i7 13650hx");
};

// Build the motherboard
void LenovoDesktopBuilder::buildMotherboard() {
    desktop->setMotherboard("Lenovo Legion 5");
};

// Build the graphics card
void LenovoDesktopBuilder::buildGraphicsCard() {
    desktop->setGraphicsCard("Nvidia RTX 4060");
};
